src/api/c++/z3++.h

```diff
@@ -385,6 +412,10 @@ namespace z3 {
         context m_ctx;
     public:
         scoped_context(Z3_context c): m_ctx(c) {}
+        scoped_context(scoped_context const &) = default;
```

Comment:
these are useless

Reply:
This is a rule of 5 violation. Since a member variable is a `context`, i.e. not a singular primitive, not including them would lead to the copy operator / constructor to not be created, due to the explicit definitions of the corresponding move functions.

Reply:
Although now that `context` is marked as delete, these probably should be as well. I'll update them now.